android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt

```diff
@@ -399,11 +407,24 @@ class StripeTerminalReactNativeModule(reactContext: ReactApplicationContext) :
             }
         }
 
+        val offlineBehaviorParam = if (amount > offlineModeTransactionLimit
+            || (terminal.offlineStatus.sdk.offlinePaymentAmountsByCurrency[currency]?.toInt() ?: 0) > offlineModeStoredTransactionLimit
+        ) { OfflineBehavior.REQUIRE_ONLINE } else {
+            offlineBehavior.let {
+                when (it) {
+                    "prefer_online" -> OfflineBehavior.PREFER_ONLINE
+                    "require_online" -> OfflineBehavior.REQUIRE_ONLINE
+                    "force_offline" -> OfflineBehavior.FORCE_OFFLINE
+                    else -> OfflineBehavior.PREFER_ONLINE
+                }
+            }
+        }
```

Comment:
This is meant to be implemented within the example app, not within the Terminal SDK. Please hoist this into the example application.